contracts/LibSubmarineSimple.sol

```diff
@@ -0,0 +1,195 @@
+pragma solidity ^0.4.24;
+
+import "./SafeMath.sol";
+import "./SafeMath32.sol";
+import "./proveth/ProvethVerifier.sol";
+
+contract LibSubmarineSimple is ProvethVerifier {
+
+    using SafeMath for uint256;
+    using SafeMath32 for uint32;
+
+
+    ////////////
+    // Events //
+    ////////////
+
+    event Unlocked(
+        bytes32 indexed _commitId,
+        uint256 _commitValue
+    );
+    event Revealed(
+        bytes32 indexed _commitId,
+        uint256 _commitValue,
+        bytes32 _witness,
+        bytes32 _commitBlockHash,
+        address submarineAddr
+    );
+
+    /////////////
+    // Storage //
+    /////////////
+
+    uint8 public vee = 27; // the ECDSA v parameter: 27 allows us to be broadcast on any network (i.e. mainnet, ropsten, rinkeby etc.)
+    uint32 public commitPeriodLength; // How many blocks must a submarine be committed for before being revealed
+
+    mapping(bytes32 => CommitData) public commitData; // stored "session" state information
+
+    // A submarine send is considered "finished" when: revealed and unlocked are both true, and the amount for the unlock is greater than or equal to the reveal amount.
+    struct CommitData {
+        bool revealed; // whether the submarine commitment for this commitData instance has been revealed yet. 
+        bool unlocked; // whether the submarine commitment for this commitData instance has been unlocked yet.
+        uint256 amountRevealed; // amount the reveal transaction revealed would be sent in wei.
+        uint256 amountUnlocked; // amount the unlock transaction recieved in wei.
+    }
```

Comment:
Could save 40k gas by removing bool from CommitData and only using uint128 for amounts.

`require(revealed == false)` becomes `require(amountRevealed != 0)`.

This is in line with libsubmarine not being used for non-monetary messages.

Reply:
Yup, 202616 vs 242621. Personally I think keeping bools is simpler for a human to understand, but gas costs win out, thanks for the observation